### ISSUE TITLE
way forward for CSV.Rows

### ISSUE THREAD

**Issue by JeffreySarnoff** [JuliaData/CSV.jl#452]
This addition is very helpful both practically and for its introduction as a legit aspect of Tabular API.
So, I am hopeful that, here, this will get the ability to return fields in their respective column types, as that adds flexible utility.   And that the next front is `Tables.jl`.  Thanks for the effort.

**Comment by quinnj** [comment]
`CSV.Rows` already implements the Tables.jl API, so there shouldn't be any problem iterating csv rows and piping the output to any other Tables.jl sink. As for column types, I don't plan on supporting that any time soon with `CSV.Rows`; the very low-memory footprint of `CSV.Rows` is achieved by treating each cell on each csv row as a `String`. It gets really hairy when you try to assume types because, for example, a column might start out looking like `Int64` values, then later there might be `Float64` values, then later you might encounter a `missing` value, and then finally there might be errant `String` data near the end of the file. It's not helpful/practical to treat those early values as `Int64`, if you know that ultimately this column of values includes strings. Hence, it's impossible to know the correct type of a column without doing a full scan of a file first, which can defeat the performance gains/aim of providing a `CSV.Rows` object that can literally provide a way to do a single pass over an entire file. 

So in short, if you need type information, use `CSV.File`; if you want a low-memory footprint way to iterate thru an entire file (possibly as a more "production"-level use-case), then use `CSV.Rows` and only parse the values you need into specific types to do calculations.

Let me know if that helps or if you have any other questions/concerns.

**Comment by JeffreySarnoff** [comment]
thank you
